### PR TITLE
add option for max connection

### PR DESCRIPTION
**Pull request by zyfdegh**
add option in config file

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -274,6 +280,12 @@ func (app *App) makeServer(addr string, handler *http.Handler) (*http.Server, er
 }
 
 func (app *App) handleWS(w http.ResponseWriter, r *http.Request) {
+	if app.options.MaxConnection != 0 {
+		if connection >= app.options.MaxConnection {
+			log.Printf("reached max connection: %d", app.options.MaxConnection)
```
Comment (yudai) [inline app/app.go]: Log messages should start with a capital letter

File: app/app.go [hunk 2]
```diff
@@ -28,6 +28,10 @@ import (
 	"github.com/yudai/umutex"
 )
 
+var (
+	connection int = 0
```
Comment (yudai) [inline app/app.go]: This variable should be a member of the App struct.

File: app/app.go [hunk 3]
```diff
@@ -344,6 +356,9 @@ func (app *App) handleWS(w http.ResponseWriter, r *http.Request) {
 	}
 	log.Printf("Command is running for client %s with PID %d (args=%q)", r.RemoteAddr, cmd.Process.Pid, strings.Join(argv, " "))
 
+	connection++
+	log.Printf("Connection: %d/%d\n", connection, app.options.MaxConnection)
```
Comment (yudai) [inline app/app.go]: Should show when MaxConnection is not 0

**Comment by yudai** [comment]
Hi, thank you for the PR. I think MaxConnection makes sense and is useful.
I put some comments to get your PR to be merged.

**Comment by zyfdegh** [comment]
Thanks for your consideration of my PR. I've rearranged log and moved var `connection` as you commented. But I think it's better to log into a separate line, it's more readable.

**Comment by yudai** [comment]
@zyfdegh Thanks for the update.
I want to keep the log in a single line.
* If the log for a single event is separated into multiple lines, it's hard to track events, especially gotty receives two or more requests in a small duration
* The current number of connections is not so important information enough to give it a dedicated line  for most use cases

I therefore want to merge these two lines.

**Comment by zyfdegh** [comment]
Okay, I forgot about concurrency. I've merged now. ;-)

**Comment by yudai** [comment]
Thanks!

**Pull request merged**